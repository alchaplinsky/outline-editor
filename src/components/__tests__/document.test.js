import React from 'react'
import { shallow } from 'enzyme'

import Document from '../document'

jest.mock('shortid')

describe('Document', () => {
  describe('Document with no arguments', () => {
    it('renders correctly with no arguments', () => {
      const component = shallow(<Document />)

      expect(component).toMatchSnapshot()
    })

    it('renders empty root node', () => {
      expect(shallow(<Document />)).toHaveHTML(
        `<div class="document">
          <div class="node">
            <div class="node-self" contenteditable="true"></div>
          </div>
        </div>`
      )
    })
  })

  describe('Document with document argument passed in', () => {
    const document = {
      type: 'text',
      value: 'Document Title',
      children: [{
        type: 'text',
        value: 'First Item',
        children: []
      }]
    }

    it('should render document nodes', () => {
      const component = shallow(<Document document={document} />)

      expect(component).toMatchSnapshot()
    })

    it('renders document nodes', () => {
      expect(shallow(<Document document={document} />)).toHaveHTML(
        `<div class="document">
          <div class="node">
            <div class="node-self" contenteditable="true">Document Title</div>
            <div class="node">
              <div class="node-self" contenteditable="true">First Item</div>
            </div>
          </div>
        </div>`
      )
    })
  })
})
