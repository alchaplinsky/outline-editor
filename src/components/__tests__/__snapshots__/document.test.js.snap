// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Document with document argument passed in should render document nodes 1`] = `
<div
  className="document"
>
  <Node
    document={
      Document {
        "context": Object {},
        "props": Object {
          "document": Object {
            "children": Array [
              Object {
                "children": Array [],
                "id": "UNIQID",
                "type": "text",
                "value": "First Item",
              },
            ],
            "id": "UNIQID",
            "type": "text",
            "value": "Document Title",
          },
        },
        "refs": Object {},
        "setState": [Function],
        "state": Object {
          "caretPosition": 14,
          "focusedNode": "UNIQID",
          "node": Object {
            "children": Array [
              Object {
                "children": Array [],
                "id": "UNIQID",
                "type": "text",
                "value": "First Item",
              },
            ],
            "id": "UNIQID",
            "type": "text",
            "value": "Document Title",
          },
        },
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <Document
              document={
                Object {
                  "children": Array [
                    Object {
                      "children": Array [],
                      "id": "UNIQID",
                      "type": "text",
                      "value": "First Item",
                    },
                  ],
                  "id": "UNIQID",
                  "type": "text",
                  "value": "Document Title",
                }
              }
            />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <div
              className="document"
            >
              <Node
                document={[Circular]}
                node={
                  Object {
                    "children": Array [
                      Object {
                        "children": Array [],
                        "id": "UNIQID",
                        "type": "text",
                        "value": "First Item",
                      },
                    ],
                    "id": "UNIQID",
                    "type": "text",
                    "value": "Document Title",
                  }
                }
              />
            </div>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    node={
      Object {
        "children": Array [
          Object {
            "children": Array [],
            "id": "UNIQID",
            "type": "text",
            "value": "First Item",
          },
        ],
        "id": "UNIQID",
        "type": "text",
        "value": "Document Title",
      }
    }
  />
</div>
`;

exports[`Document Document with no arguments renders correctly with no arguments 1`] = `
<div
  className="document"
>
  <Node
    document={
      Document {
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "setState": [Function],
        "state": Object {
          "caretPosition": 0,
          "focusedNode": "UNIQID",
          "node": Object {
            "children": Array [],
            "id": "UNIQID",
            "type": "text",
            "value": "",
          },
        },
        "updater": Updater {
          "_callbacks": Array [],
          "_renderer": ReactShallowRenderer {
            "_context": Object {},
            "_didScheduleRenderPhaseUpdate": false,
            "_dispatcher": Object {
              "readContext": [Function],
              "useCallback": [Function],
              "useContext": [Function],
              "useDebugValue": [Function],
              "useEffect": [Function],
              "useImperativeHandle": [Function],
              "useLayoutEffect": [Function],
              "useMemo": [Function],
              "useReducer": [Function],
              "useRef": [Function],
              "useState": [Function],
            },
            "_element": <Document />,
            "_firstWorkInProgressHook": null,
            "_forcedUpdate": false,
            "_instance": [Circular],
            "_isReRender": false,
            "_newState": null,
            "_numberOfReRenders": 0,
            "_renderPhaseUpdates": null,
            "_rendered": <div
              className="document"
            >
              <Node
                document={[Circular]}
                node={
                  Object {
                    "children": Array [],
                    "id": "UNIQID",
                    "type": "text",
                    "value": "",
                  }
                }
              />
            </div>,
            "_rendering": false,
            "_updater": [Circular],
            "_workInProgressHook": null,
          },
        },
      }
    }
    node={
      Object {
        "children": Array [],
        "id": "UNIQID",
        "type": "text",
        "value": "",
      }
    }
  />
</div>
`;
